cmd/utils/flags.go

```diff
@@ -1891,6 +1899,27 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *ethconfig.Config) {
 		if !ctx.IsSet(MinerGasPriceFlag.Name) {
 			cfg.Miner.GasPrice = big.NewInt(1)
 		}
+	case ctx.IsSet(ChainFlag.Name):
+		// initialize the chain from a genesis file
+		genesis, genesisHash, err := initCustomGenesis(ctx, stack, ctx.String(ChainFlag.Name))
+		if err != nil {
+			Fatalf("Failed to initialize genesis from file: %v", err)
+		}
+		cfg.Genesis = genesis
```

Comment:
Do we have to do anything with the data dir?
We could, say, put `if ctx.IsSet(ChainFlag.Name) && cfg.DataDir == Default { cfg.DataDir = '....' } in `SetDataDir`

Reply:
Not sure I understand the objective of this.

Reply:
It's so that a custom genesis won't overwrite the mainnet default directory

Reply:
Ok, that is not a problem because for specific chains a sub-directory is created.